scripts/cli/tests/12_B_settlement.js

```diff
@@ -0,0 +1,280 @@
+  let venueCounter = await api.query.settlement.venueCounter();
+  let nonceObj = { nonce: reqImports.nonces.get(sender.address) };
+  let venueDetails = [0];
+
+  const transaction = await api.tx.settlement.createVenue(venueDetails, [
+    sender.address,
+  ]);
+
+  const result = await reqImports.sendTransaction(
+    transaction,
+    sender,
+    nonceObj
+  );
+
+  const passed = result.findRecord("system", "ExtrinsicSuccess");
+  if (passed) reqImports.fail_count--;
+
+  reqImports.nonces.set(
+    sender.address,
+    reqImports.nonces.get(sender.address).addn(1)
+  );
+
+  return venueCounter;
+}
+
+async function addInstruction(
+  api,
+  venueCounter,
+  sender,
+  sender_did,
+  receiver_did,
+  ticker,
+  amount
+) {
+  let result;
+  let instructionCounter = await api.query.settlement.instructionCounter();
+  let nonceObj = { nonce: reqImports.nonces.get(sender.address) };
+  let leg = {
+    from: sender_did,
+    to: receiver_did,
+    asset: ticker,
+    amount: amount,
+  };
+
+    transaction = await api.tx.settlement.addInstruction(
+      venueCounter,
+      0,
+      null,
+      [leg]
+    );
+
+    result = await reqImports.sendTransaction(transaction, sender, nonceObj);
+  
+
+  const passed = result.findRecord("system", "ExtrinsicSuccess");
+  if (passed) reqImports.fail_count--;
+
+  reqImports.nonces.set(
+    sender.address,
+    reqImports.nonces.get(sender.address).addn(1)
+  );
+
+  return instructionCounter;
+}
+
+async function authorizeInstruction(api, sender, instructionCounter) {
+  let nonceObj = { nonce: reqImports.nonces.get(sender.address) };
+  const transaction = await api.tx.settlement.authorizeInstruction(
+    instructionCounter
+  );
+  const result = await reqImports.sendTransaction(
+    transaction,
+    sender,
+    nonceObj
+  );
+  const passed = result.findRecord("system", "ExtrinsicSuccess");
+  if (passed) reqImports.fail_count--;
+
+  reqImports.nonces.set(
+    sender.address,
+    reqImports.nonces.get(sender.address).addn(1)
+  );
+}
+
+async function unauthorizeInstruction(api, sender, instructionCounter) {
+  let nonceObj = { nonce: reqImports.nonces.get(sender.address) };
+  const transaction = await api.tx.settlement.unauthorizeInstruction(
+    instructionCounter
+  );
+  const result = await reqImports.sendTransaction(
+    transaction,
+    sender,
+    nonceObj
+  );
+  const passed = result.findRecord("system", "ExtrinsicSuccess");
+  if (passed) reqImports.fail_count--;
+
+  reqImports.nonces.set(
+    sender.address,
+    reqImports.nonces.get(sender.address).addn(1)
+  );
+}
+
+async function rejectInstruction(api, sender, instructionCounter) {
+  let nonceObj = { nonce: reqImports.nonces.get(sender.address) };
+  const transaction = await api.tx.settlement.rejectInstruction(
+    instructionCounter
+  );
+  const result = await reqImports.sendTransaction(
+    transaction,
+    sender,
+    nonceObj
+  );
+  const passed = result.findRecord("system", "ExtrinsicSuccess");
```

Comment:
This, and most other tests, could benefit from a utility function (included in my Governance v2 PR (https://github.com/PolymathNetwork/Polymesh/pull/516), in commit https://github.com/PolymathNetwork/Polymesh/pull/516/commits/96763f4821642e852598de25dbe3a716ee8f3e85), which we can move to `util/`:
```js

async function sendTx(signer, tx) {
  let nonceObj = { nonce: reqImports.nonces.get(signer.address) };
  const result = await reqImports.sendTransaction(tx, signer, nonceObj);
  const passed = result.findRecord('system', 'ExtrinsicSuccess');
  if (passed) reqImports.fail_count--;
  reqImports.nonces.set(signer.address, reqImports.nonces.get(signer.address).addn(1));
}
```

Then this function is just:
```js
async function rejectInstruction(api, signer, instructionCounter) {
  await sendTx(signer, api.tx.settlement.rejectInstruction(instructionCounter));
}
```
This should substantially reduce duplication and clarify what's special about each functions instead of drowning out.

Reply:
That is great, I would check it out thanks.